docs/contributing.rst

```diff
@@ -267,6 +267,12 @@ And their unit-tests are in::
 
     tests/test_loader.py
 
+Even though ``pytest`` supports using ``assert`` statements, do not use them
+for Scrapy tests. One of our test environments runs Scrapy tests with
+``PYTHONOPTIMIZE=1``, which removes those statements, making tests that use
+those statements pointless in that test environment.
```

Comment:
As a personal opinion, I'd not trade being able to use pytest asserts in tests for being able to automatically check PYTHONOPTIMIZE=1 case on every commit.

I'm not sure what are the practical reasons for the users to run scrapy under PYTHONOPTIMIZE=1. We could make fixes to ensure it works (if for some reason some user want it), and merge them to Scrapy, but a few minutes on CI on every commit + not being able to use asserts in tests - not sure this is justified.

Reply:
> I'm not sure what are the practical reasons for the users to run scrapy under PYTHONOPTIMIZE=1.

Scrapy may be used in an environment where `PYTHONOPTIMIZE` is also used for other package(s). But no, I don’t expect much usage of `PYTHONOPTIMIZE` with Scrapy. In Python in general, I think `PYTHONOPTIMIZE` is seldom used.

I’m OK with dropping this.